File: src/main/resources/UserDBPlugin.rb
class UserDBPlugin
  include Java::io.sato.plugin.Plugin

  def findLecturesOfUser(name, age)
    puts "Ruby Hello #{name}, #{age}"
    Hash["Calculus" => 4, "Physics" => 3, "Chemistry" => 3, "Biology" => 3]
  end

  def documentation
    'Ruby UserDBPlugin'
  end

end
UserDBPlugin.new

File: src/main/resources/UserDBPlugin.py
from io.sato.plugin import Plugin


class UserDBPlugin(Plugin):
    def findLecturesOfUser(self, name, number):
        print("Python Hello {}, {}".format(name, number))
        return {"Calculus": 4, "Physics": 3, "Chemistry": 3, "Biology": 3}

    def documentation(self):
        return "Python UserDBPlugin"
